x-pack/platform/plugins/private/snapshot_restore/public/application/lib/snapshot_list_params.ts

```diff
@@ -50,6 +50,17 @@ const resetSearchOptions = (listParams: SnapshotListParams): SnapshotListParams
   searchOperator: undefined,
 });
 
+export const escapeString = (inputString: string) => {
+  // List of characters to unescape
+  const unescapeSpecialChars = /\\([{}()\\])/g;
+
+  // List of characters to escape
+  const escapeSpecialChars = /[{}()\\]/g;
+
+  // Unescape any special char and escape again to avoid double escaping
+  return inputString.replace(unescapeSpecialChars, '$1').replace(escapeSpecialChars, '\\$&');
+};
```

Comment:
nit: I think there are a few small optimizations we could do to this code to avoid having to do two `.replace` operations which results in two full regex scans of the string. This along side a direct replacement strategy will help us avoid redundant operations in case an input string is already properly escaped, processing only characters that are incorrectly escaped instead.

```typescript
// Match either an escaped special character (\{, \}, \(, \), \\) or an unescaped special character ({, }, (, ), \)
// If it's an escaped character, remove the escape (\x -> x); otherwise, add an escape (x -> \x)
return inputString.replace(/\\([{}()\\])|([{}()\\])/g, (match, unescaped, needsEscape) => 
  unescaped ? unescaped : `\\${needsEscape}`
);
```